dockstore-common/src/main/java/io/dockstore/common/yaml/AbstractYamlService.java

```diff
@@ -0,0 +1,225 @@
+/*
+ *    Copyright 2020 OICR
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package io.dockstore.common.yaml;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * Abstract base class for services defined in .dockstore.yml
+ */
+public abstract class AbstractYamlService {
+    private String name;
+    private String author;
+    private String description;
+    private List<String> files;
+    private Scripts scripts;
+    private Map<String, EnvironmentVariable> environment;
+    private Map<String, DataSet> data;
+
+    public String getAuthor() {
+        return author;
+    }
+
+    public void setAuthor(String author) {
+        this.author = author;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public List<String> getFiles() {
+        return files;
+    }
+
+    public void setFiles(List<String> files) {
+        this.files = files;
+    }
+
+    public Scripts getScripts() {
+        return scripts;
+    }
+
+    public void setScripts(Scripts scripts) {
+        this.scripts = scripts;
+    }
+
+    public Map<String, EnvironmentVariable> getEnvironment() {
```

Comment:
This model is one of our creation, so could use javadoc particularly with the less intuitive fields like environment, data, port